sn_networking/src/record_store.rs

```diff
@@ -210,6 +211,33 @@ impl DiskBackedRecordStore {
             warn!("Record storage didn't have the distance_range setup yet.");
         }
     }
+
+    #[allow(dead_code)]
+    /// Calculate the cost to store data for our current store state
+    pub fn store_cost(&self) -> Token {
+        let records_len = self.records().count();
+
+        let mut cost = Token::from_nano(1);
+        const NANOS_PER_SNT: u64 = 1_000_000_000;
+
+        // if we imagine that at FULL capacity we want _all the money_ (ie, we never want to reach full capacity)
+        // So we get the total amount of nanos in the network,
+        // we divide that number by our MAX_RECORD_COUNT to know
+        // what to charge for each additional record
+        let step_per_record = ((2 ^ 32) * NANOS_PER_SNT) / MAX_RECORDS_COUNT as u64;
```

Comment:
Can we just use the `sn_transfers::TOTAL_SUPPLY` const ? I see is not public, perhaps it shall be public in sn_protocol or some other public crate/API.

Reply:
Ah, yes! I was looking for something like that. I only saw in sn_dbc and wanted to get something up for discussion w/o more repo faff.

Reply:
I'll update main for this.